File: demo/src/main/java/com/ml/demo/domain/Company.java
package com.ml.demo.domain;

import com.easy.query.core.annotation.Column;
import com.easy.query.core.annotation.EntityProxy;
import com.easy.query.core.annotation.Navigate;
import com.easy.query.core.annotation.Table;
import com.easy.query.core.enums.RelationTypeEnum;
import com.easy.query.core.proxy.ProxyEntityAvailable;
import com.ml.demo.domain.proxy.CompanyProxy;
import lombok.Data;
import lombok.experimental.FieldNameConstants;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * create time 2025/6/20 17:00
 * 文件说明
 *
 * @author xuejiaming
 */
@Data
@Table("t_company")
@EntityProxy
@FieldNameConstants
public class Company implements ProxyEntityAvailable<Company , CompanyProxy> {
    /**
     * 企业id
     */
    @Column(primaryKey = true)
    private String id;
    /**
     * 企业名称
     */
    private String name;

    /**
     * 企业创建时间
     */
    private LocalDateTime createTime;

    /**
     * 注册资金
     */
    private BigDecimal registerMoney;

}

File: untitled/src/main/kotlin/entity/Company.kt
package com.easy.entity

import com.easy.query.core.annotation.Column
import com.easy.query.core.annotation.EntityProxy
import com.easy.query.core.annotation.Table
import java.time.LocalDateTime


import com.easy.query.core.proxy.ProxyEntityAvailable;

import com.easy.entity.proxy.CompanyProxy;


@Table("company")
@EntityProxy
class Company : ProxyEntityAvailable<Company, CompanyProxy> {
    @Column(primaryKey = true)
    var id: String? = null;
    var name: String? = null;
    var createTime: LocalDateTime? = null;

}